### ISSUE TITLE
Automatic dependency/classpath handling with maven

### ISSUE THREAD

**Issue by mdealer** [DontShaveTheYak/groovy-guru#26]
Good work so far, this extension looks promising.

But, I have a maven Groovy project (Jenkins shared library) and there are around 14000 files in the maven repo cache with direct and indirect dependencies of various versions of Jenkins and their plugins. It would be nice if adding every JAR file would not be required and the extension would detect maven and use its repo cache and searched it for the required JARs.

**Comment by shadycuz** [comment]
Correct, It only reads jars from the parent directories. Would recursion fix the issue for you?

**Comment by mdealer** [comment]
> Correct, It only reads jars from the parent directories. Would recursion fix the issue for you?

I don't think so, because maven cache contains many different versions of everything. Versions are project specific and defined in the pom.xml.

**Comment by shadycuz** [comment]
oh I see @mdealer 

Thanks

**Comment by turchinc** [comment]
I was also struggling with this (still am!) - I am trying to setup a groovy scripting environment for my java based application framework in VS Code. 

I use gradle for dependency management. I have a working gradle groovy project that builds the code (on the command line and in vscode: `gradle compileGroovy` completes as expected) but the dependencies are not recognized in the code editor, with lots of `unable to resolve class xxx` errors for my imports.

I thought I could automatically update the `groovy.classpath` setting in my workspace `setttings.json` directly in my gradle file like this, basically adding all the gradle dependencies to the extension classpath:

```groovy

task updateGroovyClasspath  {
		doLast {
			def javaDeps = configurations.runtimeClasspath.files.collect { "$it" }
			def f = file("../.vscode/settings.json")
			def builder = new JsonBuilder(new JsonSlurper().parse(f))
			builder.content['groovy.classpath'] = javaDeps 
			f.write(builder.toPrettyString())
		}
}

compileGroovy {
		dependsOn updateGroovyClasspath
}

```

This works (at least I see lots of dependencies now in the settings), but the `unable to resolve class xxx`  errors persist. Should this work? Am I missing something? Thanks.
